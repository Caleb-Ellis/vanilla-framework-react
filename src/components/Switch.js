import React from 'react';

class Switch extends React.Component {
  render() {
    return (
      // Why are styled not being applied to this component?
      <button className="p-switch" id="switch-on" type="button" role="switch" aria-checked="true" aria-labelledby="switch-on-label">
        <span>On</span>
        <span>Off</span>
      </button>
    )
  }
}

export default Switch;
