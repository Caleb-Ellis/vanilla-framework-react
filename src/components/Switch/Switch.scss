@import 'vanilla-framework/scss/patterns_switch';
@include vf-p-switch;