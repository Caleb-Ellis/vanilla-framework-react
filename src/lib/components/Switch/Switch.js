import React from 'react';
import PropTypes from 'prop-types';

const Switch = (props) => {
  const { checked, id, label } = props;

  return (
    <label htmlFor={id}>
      {label}
      <input
        id={id}
        type="checkbox"
        className="p-switch"
        value={checked ? 'on' : 'off'}
        checked={checked}
      />
      <div className="p-switch__slider" />
    </label>
  );
};

Switch.defaultProps = {
  checked: false,
  label: undefined,
};

Switch.propTypes = {
  checked: PropTypes.bool,
  id: PropTypes.string.isRequired,
  label: PropTypes.string,
};

Switch.displayName = 'Switch';

export default Switch;
