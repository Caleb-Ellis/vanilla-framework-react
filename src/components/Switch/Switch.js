import React from 'react';
import PropTypes from 'prop-types';
import './Switch.css';

class Switch extends React.Component {
  constructor(props) {
    super(props);
    this.switchToggle = this.switchToggle.bind(this);
    this.state = {
      checked: false,
    };
  }

  switchToggle() {
    const currentState = this.state.checked;
    this.setState({ checked: !currentState });
  }

  render() {
    return (
      <span>
        <label id="switch-on-label" htmlFor="switch-on">{ this.props.label }</label>
        <button
          id="switch-on"
          className="p-switch"
          type="button"
          role="switch"
          aria-checked={this.state.checked}
          aria-labelledby="switch-on-label"
          onClick={() => this.switchToggle(this)}
        >
          <span>On</span>
          <span>Off</span>
        </button>
      </span>
    );
  }
}

Switch.propTypes = {
  label: PropTypes.string.isRequired,
};

export default Switch;
