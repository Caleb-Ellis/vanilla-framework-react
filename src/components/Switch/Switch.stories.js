import React from 'react';
import { storiesOf } from '@storybook/react';

import Switch from './Switch';

storiesOf('Switch', module)
  .add('Default', () => (
    <Switch label="On/Off" />
  ));
