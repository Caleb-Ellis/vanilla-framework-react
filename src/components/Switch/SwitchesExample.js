import React from 'react';
import Switch from './Switch';

class SwitchesExample extends React.Component {
  render() {
    return (
      <div>
        <Switch label="On/Off" />
      </div>
    );
  }
}

export default SwitchesExample;
