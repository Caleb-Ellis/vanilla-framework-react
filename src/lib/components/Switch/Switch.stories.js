import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { boolean, text } from '@storybook/addon-knobs';

import Switch from './Switch';

storiesOf('Switch', module)
  .add('Default',
    withInfo('You can use this switch pattern to display on and off content, such as for settings or simple controls. By changing the aria-checked attribute from true or false will animate the switch on/off.')(() =>
      (<Switch
        id="switch"
        checked={boolean('Checked', false)}
        label={text('Label', 'Switch')}
      />),
    ),
  );
