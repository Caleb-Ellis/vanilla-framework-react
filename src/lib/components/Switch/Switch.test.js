import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Switch from './Switch';

describe('Switch component', () => {
  it('should compare with a snapshot', () => {
    const switchComponent = ReactTestRenderer.create(
      <Switch id="switch" label="Turn On/Off" />,
    );
    const json = switchComponent.toJSON();
    expect(json).toMatchSnapshot();
  });
});

