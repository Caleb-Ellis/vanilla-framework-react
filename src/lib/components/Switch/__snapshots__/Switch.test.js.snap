// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch component should compare with a snapshot 1`] = `
<label
  htmlFor="switch"
>
  Turn On/Off
  <input
    checked={false}
    className="p-switch"
    id="switch"
    type="checkbox"
    value="off"
  />
  <div
    className="p-switch__slider"
  />
</label>
`;
